library/network/test/susefirewall_services_test.rb

```diff
@@ -9,15 +9,24 @@
 SERVICES_DATA_PATH = File.join(
   File.expand_path(File.dirname(__FILE__)),
   "data",
-  Yast::SuSEFirewallServicesClass::SERVICES_DIR
+  Yast::SuSEFirewall2ServicesClass::SERVICES_DIR
 )
 
 # Adjusts SuSEFirewallServices to read data from test-directory
 def setup_data_dir
-  stub_const("Yast::SuSEFirewallServicesClass::SERVICES_DIR", SERVICES_DATA_PATH)
+  stub_const("Yast::SuSEFirewall2ServicesClass::SERVICES_DIR", SERVICES_DATA_PATH)
 end
 
 describe Yast::SuSEFirewallServices do
+
+  before :example do
+    # We shouldn't run this test if firewalld is running
+    if Yast::SuSEFirewallServices.is_a?(Yast::SuSEFirewalldServicesClass)
+      skip "FirewallD backend is not supported by this test" do
+      end
```

Comment:
I suggest to mock behavior or runned backend. E.g. you can create mocked responses of firewalld CLI. As example see e.g. https://github.com/yast/yast-bootloader/blob/master/test/sections_test.rb#L23